Noushhhh/42_TRANSCENDANCE
# 140 setdefaultuserprofilephotoinnestjs

## Update Summary

This update includes several improvements related to file cleanup and avatar handling in the project. The changes aim to enhance project organization and ensure that default avatars are handled correctly.

### Makefile Changes

In the Makefile:
- Modified the `del_uploads` target to exclude 'defaultProfileImage.jpg' from deletion when removing files in the 'uploads' directory. This ensures that the default avatar image is preserved.

### Auth Service Changes

In `auth.service.js` and `auth.service.ts`:
- Added the import statement for the `DEFAULT_AVATAR_PATH` constant.
- Set the 'avatar' field to `DEFAULT_AVATAR_PATH` when creating a new user. This ensures that a default avatar is assigned upon user creation.

### Constants Update

In `constants.ts`:
- Defined the `DEFAULT_AVATAR_PATH` constant to specify the path to the default avatar image.

### Users Controller and Service Improvements

In `users.controller.ts` and `users.service.ts`:
- Updated avatar handling logic to prevent the deletion of the default avatar image when updating a user's avatar.
- Improved error logging and handling in the `UsersController`.

### UserProfileSetup Component Refactoring

In `UserProfileSetup.tsx`:
- Refactored the code to enhance readability.
- Added comments for better code documentation.

These changes enhance the project's file cleanup process and ensure that the default avatar is correctly assigned to new users.

## How to Test

Please verify these changes by following these steps:

1. Execute the `del_uploads` Makefile target.
2. Create a new user and confirm that the default avatar is assigned correctly.
3. Test user avatar updates and ensure the default avatar remains intact.

Your feedback on these improvements is highly appreciated!
